ibis/expr/types/pretty.py

```diff
@@ -219,21 +262,76 @@ def isnull(x):
     return out, min_width, max_width
 
 
-def format_dtype(dtype):
-    max_string = ibis.options.repr.interactive.max_string
+def format_dtype(dtype, max_string: int) -> Text:
```

Comment:
this is a small change. Before, the passed in max_string was ignored, and we only used the one in options.repr.interactive.

So `t.island.preview(max_string=5)` now gives:

```
        ┏━━━━━━━━┓
        ┃ island ┃
        ┡━━━━━━━━┩
        │ stri…  │
        ├────────┤
        │ Torg…  │
        │ Torg…  │
        │ Torg…  │
        │ …      │
        └────────┘
```

without this, you get the full `string`